spec/unit/pops/evaluator/runtime3_converter_spec.rb

```diff
@@ -30,22 +30,36 @@
     expect(converter.convert(v, {}, nil)).to equal(v)
   end
 
-  it 'converts the symbol :undef to the undef value' do
+  it 'converts top level symbol :undef to the given undef value' do
     expect(converter.convert(:undef, {}, 'undef value')).to eql('undef value')
   end
 
-  it 'converts the nil to the undef value' do
+  it 'converts top level nil value to the given undef value' do
     expect(converter.convert(nil, {}, 'undef value')).to eql('undef value')
   end
 
-  it 'convert a symbol nested in an array' do
-    expect(converter.convert({'foo' => :undef}, {}, 'undef value')).to eql({'foo' => 'undef value'})
+  it 'converts nested :undef in a hash to nil' do
+    expect(converter.convert({'foo' => :undef}, {}, 'undef value')).to eql({'foo' => nil})
```

Comment:
Is it possible to have to more deeply nested `:undef`s, something like `{'foo' => {'bar' => :undef}}`?

Reply:
yes, the converter is recursive - all :undef values at any level will be converted

Reply:
I can update this single test to wrap in a second level and in an array etc. if you think that is better